core/src/main/java/com/sekwah/advancedportals/core/tags/TriggerBlockTag.java

```diff
@@ -38,14 +44,38 @@ public String description() {
     @Override
     public List<String> autoComplete(String argData) {
         var triggerBlocks = serverContainer.getCommonTriggerBlocks()
-                                .stream()
-                                .filter(block -> block.contains(argData))
-                                .toList();
+                .stream()
+                .filter(block -> block.contains(argData))
+                .toList();
 
         if (triggerBlocks.isEmpty()) {
             return serverContainer.getAllTriggerBlocks();
         }
 
         return triggerBlocks;
     }
-}
+
+    @Override
+    public boolean created(TagTarget target, PlayerContainer player, String[] argData) {
+        for (int i = 0; i < argData.length; i++) {
+            String material = serverContainer.matchMaterialName(argData[i]);
+            if (material == null || !isValidMaterial(material)) {
+                infoLogger.warning(Lang.translate("tag.triggerblock.error.invalidmaterial"));
```

Comment:
Don't you want to tell the player what they got wrong rather than just logging it in the console?